### ISSUE TITLE
Remote API: /container/{bad id}/start returns 502

### ISSUE THREAD

**Issue by johnflavin** [docker/docker#25620]
**Output of `docker version`:**

```
Client:
 Version:      1.12.0
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   8eab29e
 Built:        Thu Jul 28 21:04:48 2016
 OS/Arch:      darwin/amd64
 Experimental: true

Server:
 Version:      1.12.0
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   8eab29e
 Built:        Thu Jul 28 21:04:48 2016
 OS/Arch:      linux/amd64
 Experimental: true
```

**Output of `docker info`:**

```
Containers: 0
 Running: 0
 Paused: 0
 Stopped: 0
Images: 81
Server Version: 1.12.0
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 95
 Dirperm1 Supported: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: overlay host bridge null
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Security Options: seccomp
Kernel Version: 4.4.16-moby
Operating System: Alpine Linux v3.4
OSType: linux
Architecture: x86_64
CPUs: 4
Total Memory: 1.954 GiB
Name: moby
ID: TDFT:QMHZ:QWMG:ZKNS:3BVF:GV75:F7HR:RL6D:GUE6:I3JC:SENA:CH6Y
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): true
 File Descriptors: 18
 Goroutines: 28
 System Time: 2016-08-11T15:00:55.060722386Z
 EventsListeners: 1
No Proxy: *.local, 169.254/16
Username: xnat
Registry: https://index.docker.io/v1/
Experimental: true
Insecure Registries:
 127.0.0.0/8
```

**Steps to reproduce the issue:**
1. Make up a container ID that doesn't exist. For example, `1`.
2. Using the remote API, POST to `/containers/{some id}/start`

**Describe the results you received:**
`HTTP/1.1 502 Bad Gateway`

**Describe the results you expected:**
`HTTP/1.1 404 Not Found`

**Comment by cpuguy83** [comment]
This looks specifically like an issue with docker4mac.
404 is correctly returned by the docker daemon.

ping @justincormack

**Comment by justincormack** [comment]
Thanks, have filed an internal issue https://github.com/docker/pinata/issues/4825 looks like our minimal proxy not behaving correctly. Hoping we can remove it soon, but will fix anyway.

**Comment by johnflavin** [comment]
Just got around to checking on this (months late). It was fixed sometime between 1.12.0 and 17.03.0-ce.

**Comment by thaJeztah** [comment]
thanks @johnflavin
